### ISSUE TITLE
[BUG] Fatal Error on Engine when using Delay Activity while using non serializable Variables

### ISSUE THREAD

**Issue by RenatoCapelo** [elsa-workflows/elsa-core#5875]
## Description
When executing a workflow that contains an assigned variable of type `byte[]` and a Delay Activity the engine will crash and will not be able to startup until the record is deleted from the boomark table on the database.

## Steps to Reproduce

1. **Detailed Steps**: 

- Create a new workflow
- Add a variable of type Byte Array
- Add a Download File Activity from the HTTP Category
- As input pass a sample file e.g: https://pdfobject.com/pdf/sample-3pp.pdf
- Bind the Output of `Response Content Bytes` to the variable created above
- After the download activity connect a Delay of 1 second
- Add an Activity after the delay e.g: Console.WriteLine
- Run the workflow

2. **Code Snippets**: 

3. **Attachments**:
   - **Workflow JSON**: [workflow-1.json](https://github.com/user-attachments/files/16559367/workflow-1.json)
   - **Sample Project**: N/a Using clean docker image of studio and server version 3.2.0-rc2.369

4. **Reproduction Rate**: Every time

5. **Video/Screenshots**: 
![image](https://github.com/user-attachments/assets/f3e5e3d0-8c1e-4339-91fd-135a84546ecf)
![image](https://github.com/user-attachments/assets/47afc44d-3e8d-4715-bc28-781cc65dec08)
![image](https://github.com/user-attachments/assets/4445f2de-db61-4fb7-90ff-acda3dff6b8f)
![image](https://github.com/user-attachments/assets/e29a88d0-c805-4cf6-b31e-dfc840ea94ff)

## Expected Behavior
The instance not crashing (and possibly being able to access the variable value?)

## Actual Behavior
Instance crashing and not recovering

## Screenshots
If possible, add screenshots or screen recordings to help explain the problem.

## Environment
- **Elsa Package Version**: 3.2.0-rc2.369
- **Operating System**: debian:12-slim. (Docker image elsaworkflows/elsa-server-and-studio-v3:latest sha256:79ca16e943ebb8d096550fda060f94d748d6651f84d209f65403868e9e8f0231)
- **Browser and Version**: N/a.

## Log Output
```
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.AggregateException: One or more errors occurred. (Failed to convert an object of type System.String to System.Byte[])
 ---> Elsa.Expressions.Exceptions.TypeConversionException: Failed to convert an object of type System.String to System.Byte[]
 ---> System.InvalidCastException: Invalid cast from 'System.String' to 'System.Byte[]'.
   at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
   at Elsa.Expressions.Helpers.ObjectConverter.ConvertTo(Object value, Type targetType, ObjectConverterOptions converterOptions) in /source/src/modules/Elsa.Expressions/Helpers/ObjectConverter.cs:line 228
   --- End of inner exception stack trace ---
   at Elsa.Expressions.Helpers.ObjectConverter.ConvertTo(Object value, Type targetType, ObjectConverterOptions converterOptions) in /source/src/modules/Elsa.Expressions/Helpers/ObjectConverter.cs:line 232
   at Elsa.Extensions.VariableExtensions.ParseValue(Variable variable, Object value) in /source/src/modules/Elsa.Workflows.Core/Extensions/VariableExtensions.cs:line 75
   at Elsa.Workflows.Services.VariablePersistenceManager.LoadVariablesAsync(WorkflowExecutionContext workflowExecutionContext) in /source/src/modules/Elsa.Workflows.Core/Services/VariablePersistenceManager.cs:line 49
   at Elsa.Workflows.Runtime.Middleware.Workflows.PersistentVariablesMiddleware.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/PersistentVariablesMiddleware.cs:line 25
   at Elsa.Workflows.Runtime.Middleware.Workflows.PersistWorkflowExecutionLogMiddleware.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/PersistWorkflowExecutionLogMiddleware.cs:line 15
   at Elsa.Workflows.Runtime.Middleware.Workflows.PersistActivityExecutionLogMiddleware.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/PersistActivityExecutionLogMiddleware.cs:line 32
   at Elsa.Workflows.Runtime.Middleware.Workflows.PersistBookmarkMiddleware.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/PersistBookmarkMiddleware.cs:line 22
   at Elsa.Workflows.Runtime.Middleware.Workflows.ScheduleBackgroundActivitiesMiddleware.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/ScheduleBackgroundActivitiesMiddleware.cs:line 25
   at Elsa.Workflows.Runtime.Middleware.Workflows.ExecuteDeferredActivityTasks.InvokeAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Runtime/Middleware/Workflows/ExecuteDeferredActivityTasks.cs:line 11
   at Elsa.Workflows.Pipelines.WorkflowExecution.WorkflowExecutionPipeline.ExecuteAsync(WorkflowExecutionContext context) in /source/src/modules/Elsa.Workflows.Core/Pipelines/WorkflowExecution/WorkflowExecutionPipeline.cs:line 38
   at Elsa.Workflows.Services.WorkflowRunner.RunAsync(WorkflowExecutionContext workflowExecutionContext) in /source/src/modules/Elsa.Workflows.Core/Services/WorkflowRunner.cs:line 187
   at Elsa.Workflows.Services.WorkflowRunner.RunAsync(WorkflowGraph workflowGraph, WorkflowState workflowState, RunWorkflowOptions options, CancellationToken cancellationToken) in /source/src/modules/Elsa.Workflows.Core/Services/WorkflowRunner.cs:line 162
   at Elsa.Workflows.Runtime.Services.LocalWorkflowClient.RunInstanceAsync(RunWorkflowInstanceRequest request, CancellationToken cancellationToken) in /source/src/modules/Elsa.Workflows.Runtime/Services/LocalWorkflowClient.cs:line 76
   at Elsa.Scheduling.Tasks.ResumeWorkflowTask.ExecuteAsync(TaskExecutionContext context) in /source/src/modules/Elsa.Scheduling/Tasks/ResumeWorkflowTask.cs:line 36
   at Elsa.Scheduling.Handlers.RunScheduledTaskHandler.HandleAsync(RunScheduledTask command, CancellationToken cancellationToken) in /source/src/modules/Elsa.Scheduling/Handlers/RunScheduledTaskHandler.cs:line 26
   at Elsa.Mediator.CommandStrategies.DefaultStrategy.ExecuteAsync[TResult](CommandStrategyContext context) in /source/src/common/Elsa.Mediator/CommandStrategies/DefaultStrategy.cs:line 21
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at Elsa.Mediator.Middleware.Command.Components.CommandHandlerInvokerMiddleware.InvokeAsync(CommandContext context) in /source/src/common/Elsa.Mediator/Middleware/Command/Components/CommandHandlerInvokerMiddleware.cs:line 54
   at Elsa.Mediator.Middleware.Command.CommandPipeline.InvokeAsync(CommandContext context) in /source/src/common/Elsa.Mediator/Middleware/Command/CommandPipeline.cs:line 29
   at Elsa.Mediator.Services.DefaultMediator.SendAsync(ICommand command, ICommandStrategy strategy, CancellationToken cancellationToken) in /source/src/common/Elsa.Mediator/Services/DefaultMediator.cs:line 65
   at Elsa.Mediator.Services.DefaultMediator.SendAsync(ICommand command, CancellationToken cancellationToken) in /source/src/common/Elsa.Mediator/Services/DefaultMediator.cs:line 57
   at Elsa.Scheduling.ScheduledTasks.ScheduledSpecificInstantTask.<Schedule>b__8_0(Object _, ElapsedEventArgs _) in /source/src/modules/Elsa.Scheduling/ScheduledTasks/ScheduledSpecificInstantTask.cs:line 58
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
```

**Comment by sfmskywalker** [comment]
Fixed via #5878

**Comment by sfmskywalker** [comment]
PS. For larger byte arrays, you may want to switch to using the Memory storage driver for the workflow variable, or store the downloaded file in e.g. Azure Storage and save the URL to that resource in a workflow variable.
